docs/DeveloperGuide.md

```diff
@@ -1,37 +1,176 @@
+The topic is selected by calling `TopicList#get` and `TopicList#hasAttempted` returns the attempted status.
+
+Step 3:
+Depending on the number of parameters,
+if there is 1 parameter (ie get all solutions):
+`QuestionsList#getAllSolutions` will get all solutions and `ui#printAllSolutions` will print them.
+else if there are 2 parameters (ie get one solution):
+`QuestionsList#getOneSolution` will get the specified solution and `ui#printOneSolution` will print it.
+
+### Explain feature
+The explain feature either prints the explanation to 1 question or all questions in 1 topic.
+
+The explain feature is facilitated by `Parser#processExplanationCommand`, which is called by `Parser#parseCommand`
+
+Step 1: After user runs the program and keys in the user command, the command will be passed to
+`Parser#parseCommand`.
+
+> **NOTE:** The command must contain the `explain` keyword.
+> 
+> Sequence diagram for `explain` is similar to the [sequence diagram](#solution-feature) for `solution` feature
+
+Step 2a: `Parser#processExplainCommand` first checks the number of parameters in the user command
+by calling `Parser#checkIfTwoParameters`.
+The, further processing of parameters is done by calling `Parser#getTopicOrQuestionNum`.
+This is facilitated by calling `QuestionsListByTopic#getQuestionSet` to get all questions in the specified topic.
+
+Step 2b: Before getting the explanation(s), the program first verifies if the topic has been attempted before.
+The program only prints them only if the topic is attempted before.
+The topic is selected by calling `TopicList#get` and `TopicList#hasAttempted` returns the attempted status.
+
+Step 3:
+Depending on the number of parameters,
+if there is 1 parameter (ie get all explanations):
+`QuestionsList#getAllExplanations` will get all explanations and `ui#printAllExplanations` will print them.
+else if there are 2 parameters (ie get one explanation):
+`QuestionsList#getOneExplanation` will get the specified explanation and `ui#printOneExplanation` will print it.
+
 
 ## Product scope
+
 ### Target user profile
 
-{Describe the target user profile}
+CS2113/2103 students who wish to have an efficient tool to help them revise relevant concepts.
 
 ### Value proposition
 
-{Describe the value proposition: what problem does it solve?}
+Integration of key notions and learning learning objectives for CS2113 course, self-adaptive testing for understanding and feedback.
 
 ## User Stories
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+| Version | As a ...                       | I want to ...                                        | So that I can ...                                                      |
```

Comment:
maybe you could categorise the user stories by priorities and include their priority into the userstories table